src/html/electronpreview.html

```diff
@@ -0,0 +1,559 @@
+                </section>
+                <section>
+                    <p><a id="activeLinks10" href="#"></a></p>
+                </section>
+                </article>  <!-- closing content column -->
+
+            </div>  <!-- closing main div -->
+        </section>
+    </div>
+
+<script>
+    var keys = [];
+    var messages = [];
+    var currentMsg;
+    $(document).ready(function () {
+        // populate the preview's text.
+        fetch_and_insert_text();
+        setVideoUrl(false);
+    });
+
+    // Parses the query string and returns an object container key-value pairs from
+    // decoded querystring variables
+    function getQueryStringKeys() {
+        var queryKvps = {};
+        var kvps = window.location.search.substring(1).split("&");
+        for (var i = 0; i < kvps.length; i++) {
+            var kvp_pieces = kvps[i].split("=");
+            queryKvps[decodeURIComponent(kvp_pieces[0])] = decodeURIComponent(kvp_pieces[1]);
+        }
+        return queryKvps;
+    }
+
+        // Parses the "lang" variable out of the querystring, returning the languageCode to use
+        // for the page text
+        function getLanguageCode() {
+            var languageCode = "en-US"; //default to english
+            var queryKvps = getQueryStringKeys();
+            if (queryKvps.lang && queryKvps.lang != "undefined") {
+                languageCode = queryKvps["lang"];
+            }
+            return languageCode;
+        }
+
+        // Fetch the language text file based on lang Query String variable, and pass
+        // the returned json data to the function that will place strings in the UI
+        function fetch_and_insert_text() {
+            var langaugeCode = getLanguageCode();
+            var url = '../messages/electronPreviewMessages_' + langaugeCode + ".json";
+            $.ajax({
+                type: 'GET',
+                url: url,
+                dataType: 'json',
+                success: function (data) {
+                    insertText(data);
+                    setUpTts(data);
+                },
+                error: function (jqXHR, textStatus, errorThrown) {
+                    console.log("error occured: " + textStatus);
+                }
+            });
+        }
+
+    // Youtube will auto load the captions in the language if they exist, it will not load
+    // auto generated captions when using cc_load_policy, so choose a video that supports
+    // all needed languages w/ CC.
+    function setVideoUrl(showCaptions) {
+        var youtube_url = "https://www.youtube.com/embed/yBrp8uvNAhI?disablekb=0&start=2";
+
+        if (showCaptions) {
+            var languageCode = getLanguageCode().substring(0, 2);
+            youtube_url = youtube_url + "&cc_load_policy=1" + "&cc_lang_pref=" + encodeURIComponent(languageCode);
+        }
+
+        $("#yt_holder iframe").remove();
+        $('<iframe width="560" height="315" frameborder="0" allowfullscreen></iframe>')

```

Comment:
This hard-coded markup and dimensions need to be broken out into some configurable options. Preferably all of this material should be structured as Infusion components